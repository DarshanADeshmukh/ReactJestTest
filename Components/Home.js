import React,{Component} from 'react';
import {Platform, StyleSheet, Text, View} from 'react-native';

export default class Home extends Component<Props>{

  constructor(props){
    super(props);
    this.state = {
      data:'result',
      header:'',
    }
  }

  render(){
    return(
      <View style = {styles.container}>
      <Text style = {styles.textStyle}> Home Screen </Text>
      </View>
    )
  }
  sum(a,b){
    return a + b;
  }
  substract(a,b){
    return a -b;
  }
  multiply(a,b){
    return a * b;
  }
  division(a,b){
    return a / b;
  }
  changeComponentState(x){
    this.setState({data:x * 6});
  }
  changeComponentStateHeader(name){
    this.setState({header:name});
  }

}

const styles = StyleSheet.create({
  container:{
    flex:1,
    justifyContent:'center',
    alignItems:'center',
  },
  textStyle:{
    fontSize:30,
    color:'black',
  },

});
