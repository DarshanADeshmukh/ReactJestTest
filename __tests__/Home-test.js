import 'react-native';
import React from 'react';
import Home from '../Components/Home';

// Note: test renderer must be required after react-native.
import renderer from 'react-test-renderer';

test('Home Snapshot Testing',() => {
  const homeSnap = renderer.create(
    <Home />
  )
  expect(homeSnap).toMatchSnapshot();
})
