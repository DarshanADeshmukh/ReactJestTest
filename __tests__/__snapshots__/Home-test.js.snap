// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home Snapshot Testing 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "flex": 1,
      "justifyContent": "center",
    }
  }
>
  <Text
    style={
      Object {
        "color": "black",
        "fontSize": 30,
      }
    }
  >
     Home Screen 
  </Text>
</View>
`;
